File: Courzelo-backend-LMS/Courzelo/src/main/java/org/example/courzelo/dto/responses/institution/TimetableResponse.java
package org.example.courzelo.dto.responses.institution;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.courzelo.models.institution.Timeslot;

import java.util.Date;
import java.util.List;
import java.util.Map;
@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class TimetableResponse {
    private Map<String, List<Timeslot>> groupTimetables;
    private Map<String, List<Timeslot>> teacherTimetables;
    private Date timetableWeek;

}

File: courzelo-dashboard-LMS-main/src/app/shared/models/institution/TimetableResponse.ts
export interface Timeslot {
    group: string;
    teacher: string;
    module: string;
    dayOfWeek: string; // e.g., "Monday", "Tuesday"
    startTime: string; // e.g., "09:00 AM"
    endTime: string;   // e.g., "11:00 AM"
}

export interface TimetableResponse {
    groupTimetables: { [key: string]: Timeslot[] };
    teacherTimetables: { [key: string]: Timeslot[] };
    timetableWeek: Date;
}
